pkg/pod-autoscaler/pkg/pod-resource-updater/controller.go

```diff
@@ -134,43 +134,66 @@ func (c *Controller) runNodeScaleWorker() {
 				return
 			}
 
+			isReady := false
+			for _, c := range pod.Status.Conditions {
+				if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+					isReady = true
+				}
+			}
+
+			if !isReady {
+				klog.Infof("Pod %s/%s not ready", pod.Namespace, pod.Name)
+				return
+			}
+
 			newPod, err := syncPod(pod, *podScale)
 			if err != nil {
 				klog.Error("Error syncing the pod: ", err)
 				return
 			}
 
-			// try both updates in dry-run first and then actuate them consistently
-			updatedPod, updatedPodScale, err := c.AtomicResourceUpdate(newPod, podScale)
-
-			if err != nil {
-				klog.Error("Error while updating pod and podscale: ", err)
-				//TODO: We are using this channel as a workqueue. Why don't use one?
-				c.in <- nodeScale
-				return
+			requireUpdate := false
+			for i := range pod.Spec.Containers {
+				if pod.Spec.Containers[i].Resources.Requests.Cpu().MilliValue() !=
+					newPod.Spec.Containers[i].Resources.Requests.Cpu().MilliValue() {
+					requireUpdate = true
+				}
 			}
 
-			//TODO: handle error
-			_ = c.log.Log(updatedPodScale)
+			if requireUpdate {
+				// try both updates in dry-run first and then actuate them consistently
+				updatedPod, updatedPodScale, err := c.AtomicResourceUpdate(newPod, podScale)
+
+				if err != nil {
+					klog.Error("Error while updating pod and podscale: ", err)
+					//TODO: We are using this channel as a workqueue. Why don't use one?
+					c.in <- nodeScale
+					return
+				}
+
+				//TODO: handle error
+				_ = c.log.Log(updatedPodScale)
+
+				klog.Info("Desired resources:", updatedPodScale.Spec.DesiredResources)
+				klog.Info("Capped resources:", updatedPodScale.Status.CappedResources)
+				klog.Info("Actual resources:", updatedPodScale.Status.ActualResources)
+				klog.Info("Pod resources:", updatedPod.Spec.Containers[0].Resources)
+			}
 
-			klog.Info("Desired resources:", updatedPodScale.Spec.DesiredResources)
-			klog.Info("Capped resources:", updatedPodScale.Status.CappedResources)
-			klog.Info("Actual resources:", updatedPodScale.Status.ActualResources)
-			klog.Info("Pod resources:", updatedPod.Spec.Containers[0].Resources)
 		}
 	}
 }
 
 // AtomicResourceUpdate updates a Pod and its PodScale consistently in order to keep synchronized the two resources. Before performing the real update
 // it runs a request in dry-run and it checks for any potential error
 func (c *Controller) AtomicResourceUpdate(pod *corev1.Pod, podScale *v1beta1.PodScale) (*corev1.Pod, *v1beta1.PodScale, error) {
-	var err error
-	_, _, err = c.updateResources(pod, podScale, true)
-
-	if err != nil {
-		klog.Error("Error while performing dry-run resource update: ", err)
-		return nil, nil, err
-	}
+	//var err error
```

Comment:
Why this is commented? It is a "safety  check" before applying the changes. If we want to remove it (and I agree with this decision since the `dry-run` is made server side on the `apiserver`) please delete the code. Actually we are performing two calls and the first one is useless in some way since if something goes wrong we are already catching the error.

Reply:
solved